ci-operator/step-registry/rosa/conf/machine-pool/rosa-conf-machine-pool-commands.sh

```diff
@@ -0,0 +1,148 @@
+ROSA_VERSION=$(rosa version)
+ROSA_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/ocm-token")
+if [[ ! -z "${ROSA_TOKEN}" ]]; then
+  echo "Logging into ${ROSA_LOGIN_ENV} with offline token using rosa cli ${ROSA_VERSION}"
+  rosa login --env "${ROSA_LOGIN_ENV}" --token "${ROSA_TOKEN}"
+  if [ $? -ne 0 ]; then
+    echo "Login failed"
+    exit 1
+  fi
+else
+  echo "Cannot login! You need to specify the offline token ROSA_TOKEN!"
+  exit 1
+fi
+
+# Switches
+LABELS="prowci=true"
+if [[ ! -z "$MP_LABELS" ]]; then
+  LABELS="${LABELS},${MP_LABELS}"
+fi
+
+TAINTS="prowci=true:NoSchedule"
+if [[ ! -z "$MP_TAINTS" ]]; then
+  TAINTS="${TAINTS},${MP_TAINTS}"
+fi
+
+MP_NODES_SWITCH=""
+if [[ "$MP_ENABLE_AUTOSCALING" == "true" ]]; then
+  MP_NODES_SWITCH="--enable-autoscaling --min-replicas ${MP_MIN_REPLICAS} --max-replicas ${MP_MAX_REPLICAS}"
+else
+  MP_NODES_SWITCH="--replicas ${MP_REPLICAS}"
+fi
+
+TUNING_CONFIG_SWITCH=""
+if [[ "$USE_TUNING_CONFIG" == "true" ]]; then
+  tuning_config_file=$(cat "${SHARED_DIR}/tuning_config_file")
+  TUNING_CONFIG_SWITCH="--tuning-configs ${tuning_config_file}"
+fi
+
+SPOT_INSTANCES_SWITCH=""
+if [[ "$USE_SPOT_INSTANCES" == "true" ]]; then
+  SPOT_INSTANCES_SWITCH="--use-spot-instances --spot-max-price ${SPOT_MAX_PRICE}"
+fi
+
+LOCAL_ZONE_SWITCH=""
+if [[ "$LOCAL_ZONE" == "true" ]]; then
+  LOCAL_ZONE_SWITCH=""
+  private_subnet_id=$(cat ${SHARED_DIR}/private_subnet_ids | tr -d "[']" | cut -d ',' -f 1)
+  if [[ -z "${private_subnet_id}" ]]; then
+    echo -e "The private_subnet_id is mandatory."
+    exit 1
+  fi
+
+  LOCAL_ZONE_SWITCH="--subnet ${private_subnet_id}"
+fi
+
+# Create machine pool on the cluster
+function createMachinepool() {
+  subfix=$(openssl rand -hex 2)
+  MP_NAME="mp-$subfix"
+  
+  ZONE=$1
+  AZ_SWITCH=""
+  if [[ ! -z "$ZONE" ]]; then
+    AZ_SWITCH="--availability-zone ${ZONE}"
+  fi
+
+  AUTO_REPAIR_SWITCH=""
+  if [[ "$HOSTED_CP" == "true" ]]; then
+    AUTO_REPAIR_SWITCH="--autorepair"
+  fi
+
+  echo -e " Create machine pool ${MP_NAME} on the cluster ${CLUSTER_ID} ...
+rosa create machinepool -y \
+-c ${CLUSTER_ID}
+--name ${MP_NAME} \
+--instance-type ${MP_MACHINE_TYPE} \
+--labels ${LABELS} \
+--taints ${TAINTS} \
+${MP_NODES_SWITCH} \
+${TUNING_CONFIG_SWITCH} \
+${SPOT_INSTANCES_SWITCH} \
+${AZ_SWITCH} \
+${LOCAL_ZONE_SWITCH} \
+${AUTO_REPAIR_SWITCH}
+"
+
+rosa create machinepool -y \
+                        -c ${CLUSTER_ID} \
+                        --name ${MP_NAME} \
+                        --instance-type ${MP_MACHINE_TYPE} \
+                        --labels ${LABELS} \
+                        --taints ${TAINTS} \
+                        ${MP_NODES_SWITCH} \
+                        ${TUNING_CONFIG_SWITCH} \
+                        ${SPOT_INSTANCES_SWITCH} \
+                        ${AZ_SWITCH} \
+                        ${LOCAL_ZONE_SWITCH} \
+                        ${AUTO_REPAIR_SWITCH}
+} 
+
+
+if [[ ! -z "$MP_ZONE" ]]; then
+  MP_ZONE="${CLOUD_PROVIDER_REGION}${MP_ZONE}"
```

Comment:
why combine MP_ZONE with ${CLOUD_PROVIDER_REGION}${MP_ZONE}? will it be like 'us-west-2us-west-2a'? Seems there are different for MP_ZONE for the env variable and the function variable.

Reply:
MP_ZONE is `a`, `b`,`c`